datalad/support/sshconnector.py

```diff
@@ -423,7 +423,11 @@ def __init__(self, ctrl_path, sshri, **kwargs):
         ]
         self.ctrl_path = Path(ctrl_path)
         self._opened_by_us = False
-        self._lock = threading.Lock()
+        # used by @fasteners.locked
+        self._lock = [
+            threading.Lock(),
+            fasteners.process_lock.InterProcessLock(self.ctrl_path.with_suffix('.lck'))
+        ]
```

Comment:
Oh, that's fancy.  Thanks for leaving the comment to point future confused readers in the right direction.